test/integration/framework/perf_utils.go

```diff
@@ -119,7 +119,7 @@ func (p *IntegrationTestNodePreparer) PrepareNodes(nextNodeIndex int) error {
 func (p *IntegrationTestNodePreparer) CleanupNodes() error {
 	// TODO(#93794): make CleanupNodes only clean up the nodes created by this
 	// IntegrationTestNodePreparer to make this more intuitive.
-	nodes, err := GetReadySchedulableNodes(p.client)
```

Comment:
curious why this change was needed?

Reply:
When creating nodes with taints, all these nodes are unschedulable to normal pods, so we'll always return error after calling `Filter()`. That means no node satisfied.

`checkWaitListSchedulableNodes` will return all nodes with `spec.unschedulable` is false, the only difference here is we'll patch more nodes than before, who will not be scheduled pods. So I think it's ok.

Reply:
This actually impacts all existing tests' behavior to skip nodes' health check. I'm a bit concerned on this. Can you add a param in `workloadTemplate` to make the behavior of "ignoring node health check or not" an option?

Reply:
Should we make the parameter ignore taints, instead of ignoring isNodeSchedulable?

Reply:
although... why did we even care about getting Ready nodes in the first place? I don't think we have any test cases with non-ready nodes.

Reply:
Add a param is one approach, but try to figure out the reason why we want to filter out taint nodes here. They're schedulable actually if pod tolerates the taints. 

 I think the change won't impact present tests for we have no taint-related or node-not-ready cases. But in theory, filtering out non-ready nodes when updating patches makes sense to me for the sake of performance especially we have thousands of nodes.

Another question here is when cleaning up, we only remove the `readySchedulableNodes` nodes, but we should remove all nodes created by itself. I used to submit a PR to fix this here https://github.com/kubernetes/kubernetes/pull/112237.

So in summary:
1) When calling `PrepareNodes`, I'll get ready and schedulable nodes including nodes with taints
2) When calling `CleanupNodes`, I'll remove all nodes created by itself as https://github.com/kubernetes/kubernetes/pull/112237.

WDYT?

Reply:
> I think we can prepare and cleanup all nodes, given that we always create them as ready (there are no kubelets running in our tests).

I'm ok with this. Also looking forward to @Huang-Wei 's advices.

Reply:
Created a separate PR here https://github.com/kubernetes/kubernetes/pull/113615 to make this a pure one, only focus on the API part.